drift.py

```diff
@@ -1,146 +1,90 @@
-                    # Check if the package version is different.
-                    if self.last[spec][0][name]["version"] != concrete_spec[0][name]["version"]:
-                        result.tags += ["pkg-updated"]
-                    # +---------------------+
-                    # |     Dependencies    |
-                    # +---------------------+
-                    result.tags += set(self.tag_deps(concrete_spec, spec, name))
-                    # +---------------------+
-                    # |      Variants       |
-                    # +---------------------+
-                    result.tags += set(self.tag_variants(concrete_spec, spec, name))
-                    # Send Inflection Point
-                    send(result)
-                # Save concrete spec hash as last hash.
-                self.last[spec] = concrete_spec
-            else:
-                # If the spec doesn't concretize properly we also want
-                # to record the commit at which this occurred.
-                result.tags += ["concretizaton-failed"]
+        for abstract_spec in self.specs:
+            # Separate out spec version and name.
+            name = abstract_spec
+            version = ""
+            if "@" in abstract_spec:
+                name, version = abstract_spec.split("@", 1)
+            # Check that version exists in package.py file.
+            out = run(f"spack versions --safe-only {abstract_spec}")
+            # Don't attempt to concretize if the version doesn't yet exist.
+            if version in out.stdout:
+                out = run(f"spack spec --yaml {abstract_spec}")
+                # If concretization successful check the resulting concrete specs.
+                if out.returncode == SUCCESS:
+                    concrete_spec = spack.spec.Spec().from_yaml(out.stdout)
+                    # If the spec is the same move on.
+                    if abstract_spec not in self.last:
+                        self.last[abstract_spec] = concrete_spec
+                    if self.last[abstract_spec] == concrete_spec:
+                        continue
+                    # If the spec is different check what is different and save as tags.
+                    diff = compare_specs(
+                        self.last[abstract_spec],
+                        concrete_spec,
+                        colorful=False)
+                    # Flatten list to tags with context.
+                    tags = ["%s:%s" %(x[0],x[1]) for x in diff['b_not_a']]
+                    # Save concrete spec as last spec.
+                    self.last[abstract_spec] = concrete_spec
+                # Mark failing concretization points.
+                else:
+                    tags = ["concretization-failed"]
+                # Construct Result
+                result = Result(
+                    name,
+                    version,
```

Comment:
So yeah here, I think you want to return a result that contains the old concrete spec, the new concrete spec, tags from the diff (probably including *everything* that changed -- not just stuff that was added.

Can the drift server handle indexing by full specs like spack mon can?  I think it would be really good if it could.

Reply:
Hey! Should we upload both the old spec and the new spec? Shouldn't the old spec be known already by querying the previous inflection point? Just want to make sure we're not duplicating data.